hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstantTimeGenerator.java

```diff
@@ -33,14 +35,27 @@
  */
 public class HoodieInstantTimeGenerator {
   // Format of the timestamp used for an Instant
-  private static final String INSTANT_TIMESTAMP_FORMAT = "yyyyMMddHHmmss";
-  private static final int INSTANT_TIMESTAMP_FORMAT_LENGTH = INSTANT_TIMESTAMP_FORMAT.length();
+  public static final String SECS_INSTANT_TIMESTAMP_FORMAT = "yyyyMMddHHmmss";
+  public static final int SECS_INSTANT_ID_LENGTH = SECS_INSTANT_TIMESTAMP_FORMAT.length();
+  private static final String MILLIS_INSTANT_TIMESTAMP_FORMAT = "yyyyMMddHHmmssSSS";
```

Comment:
why not public? also feel the constants below can be all public final. don't see harm exposing them but convenience in testing.

Reply:
Can we actually introduce `.` in the time format? it improves readability and people can easily distinguish it from the old format. Plus it will bypass the jdk bug with millisecond parsing?

```suggestion
  private static final String MILLIS_INSTANT_TIMESTAMP_FORMAT = "yyyyMMddHHmmss.SSS";
```

Reply:
@vinothchandar : any thoughts here.